mistralrs-core/src/lib.rs

```diff
@@ -234,8 +259,66 @@ impl MistralRs {
         this
     }
 
-    pub fn get_sender(&self) -> Sender<Request> {
-        self.sender.clone()
+    /// attempts to reboot the engine, if the sender (only way to communicate with
+    /// the engine) is closed
+    /// TODO: GS don't use anyhow for these errors?
+    pub fn reboot_engine(&self) -> anyhow::Result<()> {
+        tracing::info!("attempting to reboot");
+        // only start a new runtime if the reciever was closed. this implies
+        // that it was dropped, and therefore the tokio runtime is down.
+        if self.get_sender()?.is_closed() {
+            tracing::info!("sender is closed, rebooting");
+            let (tx, rx) = channel(10_000);
+            self.update_sender(tx)?;
+
+            let reboot_state = self.reboot_state.clone();
+
+            thread::spawn(move || {
+                let rt = Runtime::new().unwrap();
+                rt.block_on(async move {
+                    let mut engine = Engine::new(
+                        rx,
+                        reboot_state.pipeline.clone(),
+                        reboot_state.method,
+                        reboot_state.truncate_sequence,
+                        reboot_state.no_kv_cache,
+                        reboot_state.no_prefix_cache,
+                        reboot_state.prefix_cache_n,
+                        reboot_state.disable_eos_stop,
+                    );
+                    engine.run().await;
+                });
+            });
+            Ok(())
+        } else {
+            Err(anyhow::Error::msg("Did not reboot, sender not closed"))
+        }
+    }
+
+    /// TODO: GS don't use anyhow for these errors?
```

Comment:
Same as other comment, replace `TODO: GS` with `TODO(GS)` or something similar for easier discoverability.